pkg/importer/dataStream.go

```diff
@@ -175,6 +176,10 @@ func newDataStream(dso *DataStreamOptions, stream io.ReadCloser) (*DataStream, e
 		if err != nil {
 			return nil, errors.Wrapf(err, "unable to calculate iso file size")
 		}
+		// at that point, only if ds.size != 0 we know for sure that this is an iso file
+		if ds.Size != 0 {
```

Comment:
I must say I hate this assumption, I think we should check the actual format and not rely on the size value

Reply:
@igoihman the call to ds.isoSize (L.175) checks if this is iso file and if not return 0. So the logic exists there (look also L.628)

Reply:
what about the calls in `fileFormatSelector`? can't it be changed in the future?
Maybe we should change `ds.isoSize()` call and check if it's iso first.

Reply:
yeah.... so this is currently the implementation to decide whether it's an iso file or not (day one).
I wonder what other think? if we want to change this, it should be in a different PR as this one is about not resizing iso file only. I do not think the decision if it's an iso or not should be here.

Reply:
Yeah I don't understand why ds.isoSize() doesn't return an error when the input is not an iso. They have the check in there, but it simply logs the error and returns 0 size. I guess they wanted to differentiate between a fatal error (unable to open the stream) vs a non fatal one, unable to determine that the endpoint is an iso.

So for now I think simply checking is the size > 0 is enough.